master/internal/resourcemanagers/resource_pool.go

```diff
@@ -368,29 +369,47 @@ func (rp *ResourcePool) Receive(ctx *actor.Context) error {
 }
 
 func (rp *ResourcePool) receiveAgentMsg(ctx *actor.Context) error {
+	var agentID string
 	switch msg := ctx.Message().(type) {
+	// TODO(ilia): I hope go will have a good way to do this one day.
```

Comment:
previously I've just added an interface to get `msg.Agent` and used that. but it's really not better.

Reply:
yeah, then you need to implement interface + 3 methods? not much of a difference...

Reply:
yeah I agree. I really meant " but it's really not better.".